server/controllers/userController.js

```diff
@@ -0,0 +1,67 @@
+import * as db from '../models/swatchModel';
+import bcrypt from 'bcrypt';
+
+SALT_WORK_FACTOR = 10;
+
+export const userController = {};
+
+userController.createUser = (req, res, next) => {
+  const { username, email, password, key } = req.body;
+
+  bcrypt.hash(password, SALT_WORK_FACTOR)
+    .then(hash => {
+      const query = `
+        INSERT INTO users
+          (username, email, password, key)
+        SELECT
+          $1, $2, $3, $4
+        WHERE NOT EXISTS (
+          SELECT email
+          FROM users
+          WHERE email = $2
+        )
+      `;
+      const values = [
+        username,
+        email,
+        hash,
+        key
+      ];
+
+      db.query(query, values);
```

Comment:
Nit: For future reference - you may want to think about returning the `ID` of the insert in your query to ensure you actually have the row inserted into DB